main.lua

```diff
@@ -15,71 +15,134 @@ See the License for the specific language governing permissions and
 limitations under the License.
 
 --]]
-
-local bundle = require('luvi').bundle
+-- Convoluted boostrapping?
+local bundle = require("luvi").bundle
 loadstring(bundle.readfile("luvit-loader.lua"), "bundle:luvit-loader.lua")()
 
-local uv = require('uv')
-require('snapshot')
+-- Upvalues
+local uv = require("uv")
+local version = require("./package").version
+local log = require("log").log
+require("snapshot")
+
+-- Global setup
+_G.p = require("pretty-print").prettyPrint -- TBD: Can't this be removed? Luvi already injects the same global AFAIK
```

Comment:
luvi does not, luvit does

Reply:
Hm, looks like you're right - it's not present before this line. I'm confused as to what this does now: https://github.com/luvit/luvi/blob/afea1cd8dd5842a6b9bc368c016298fb7fc30820/src/lua/luvibundle.lua#L337 - for some reason I thought this was injecting the global into the bundled app...

Either way, I'll remove the note.

Reply:
Yeah, I don't really understand that part of Luvi either. It's dependent on both `deps/require.lua` and `deps/pretty-print.lua` existing in the bundle, though, so it's still pretty Luvit-specific.

Reply:
I would remove it from luvi, but it'd probably break things. I think the difference here is that luvi uses require and lit uses luvit-loader, so the two "requires" are not exactly the same?